test/fail_compilation/ctfe16694.d

```diff
@@ -0,0 +1,11 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/ctfe16694.d(11): Error: static variable barptr cannot be read at compile time
```

Comment:
Uh, sure. But that means we break out of the CTFE early instead of returning a `NULL` as an address...and I guess that's the more desirable option.